rewrite-core/src/test/java/org/openrewrite/GitRemoteTest.java

```diff
@@ -102,4 +104,28 @@ void parseRegisteredRemote(String cloneUrl, String origin, GitRemote.Service ser
         assertThat(remote.getOrganization()).isEqualTo(expectedOrganization);
         assertThat(remote.getRepositoryName()).isEqualTo(expectedRepositoryName);
     }
+
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      GitHub, false, github.com, org/repo, https://github.com/org/repo
+      GitHub, true, github.com, org/repo, ssh://git@github.com/org/repo.git
+      GitHub, false, https://github.com, org/repo, https://github.com/org/repo
+      GitHub, true, https://github.com, org/repo, ssh://git@github.com/org/repo.git
+      GitHub, false, scm.company.com/context/path, org/repo, https://scm.company.com/context/path/org/repo
+      GitHub, true, scm.company.com/context/path, org/repo, ssh://git@scm.company.com/context/path/org/repo.git
+      GitLab, false, gitlab.com, group/subgroup/repo, https://gitlab.com/group/subgroup/repo.git
+      GitLab, true, gitlab.com, group/subgroup/repo, ssh://git@gitlab.com/group/subgroup/repo.git
+      GitLab, false, scm.company.com/context/path, group/subgroup/repo, https://scm.company.com/context/path/group/subgroup/repo.git
+      GitLab, true, scm.company.com/context/path, group/subgroup/repo, ssh://git@scm.company.com/context/path/group/subgroup/repo.git
+      BitbucketCloud, false, bitbucket.org, org/repo, https://bitbucket.org/org/repo
+      BitbucketCloud, true, bitbucket.org, org/repo, ssh://git@bitbucket.org/org/repo.git
+      Bitbucket, false, scm.company.com/context/path, org/repo, https://scm.company.com/context/path/scm/org/repo
+      Bitbucket, true, scm.company.com/context/path, org/repo, ssh://git@scm.company.com/context/path/org/repo.git
```

Comment:
This one doesn't seem right by default for Bitbucket Data Center. It's default ssh clone port is 7999 and I'm not sure if it supports a context path for the ssh server.

Reply:
While 7999 is the standard port it is trivial to change it, same goes for a context path (reverse proxy)

I think we cannot assume 7999 and should have that as part of the origin.

But then the https and ssh origins are no longer the same.

It's a tricky situation either way, and I'm not sure what is the best way to handle it.

However, I know on the Moderne side we do assume the 7999 default so perhaps we need to just pick that as the lesser evil?